packages/block-editor/src/components/block-mover/index.native.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { first, last, partial, castArray } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { ToolbarButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { withInstanceId, compose } from '@wordpress/compose';
+
+const BlockMover = ( {
+	isFirst,
+	isLast,
+	onMoveDown,
+	onMoveUp,
+} ) => (
+	<>
+		<ToolbarButton
+			accessibilityLabel={ __( 'Move up' ) }
+			label={ __( 'Move up' ) }
+			isDisabled={ isFirst }
+			onClick={ onMoveUp }
+			icon="arrow-up-alt"
+		/>
+
+		<ToolbarButton
+			label={ __( 'Move down' ) }
+			isDisabled={ isLast }
+			onClick={ onMoveDown }
+			icon="arrow-down-alt"
+		/>
+	</>
+);
+
+export default compose(
+	withSelect( ( select, { clientIds } ) => {
+		const { getBlockIndex, getBlockRootClientId, getBlockOrder } = select( 'core/block-editor' );
+		const normalizedClientIds = castArray( clientIds );
+		const firstClientId = first( normalizedClientIds );
+		const rootClientId = getBlockRootClientId( first( normalizedClientIds ) );
+		const blockOrder = getBlockOrder( rootClientId );
```

Comment:
This seems to be only used for its length, maybe `getBlockCount` is a better selector?
On the other hand, if we're not actually looking at the results of getBlockOrder, couldn't we use the length of `normalizedClientIds` instead to find the last one?

Reply:
Actually, I just realized this is just what the web does. Do you think it'd make sense to share the select/dispatch logic and just fork the internal `BlockMover` component?

Reply:
I'm not much in favor. Surely the components have similar names but they're not used the same way at the moment in web and native, I'd rather we keep them separated.
But in general, I'm all for reusing the web code whenever possible.